### PR TITLE
APP-463: [C SDK] GPIO control slave application

### PR DESCRIPTION
**Pull request by vasner**
Add client application for GPIO control for posix "platform"

Review: @Slash32 and @AlexShev76

### CODE REVIEW

**Review by Slash32 — CHANGES_REQUESTED** [review]
@vnerozin this also needs changes to the gpiocontrol/pom.xml and other parts for integration into the bundle.

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

File: gpiocontrol/source/c/CMakeLists.txt [hunk 1]
```diff
@@ -47,9 +47,11 @@ else()
 endif()
 
 # For most embedded targets binary stripping is required.
-if(${KAA_PRODUCE_BINARY})
-    add_custom_command(TARGET ${APP_NAME} POST_BUILD
-            COMMAND ${CMAKE_OBJCOPY} -O binary ${APP_NAME} ${APP_NAME}.bin)
+if("${KAA_PLATFORM}" STREQUAL “posix”)
```
Comment (rasendubi) [inline gpiocontrol/source/c/CMakeLists.txt]: `s/“posix”/"posix"/`

File: common/c/targets/posix/target_gpio_led.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    
+    if (id >= NUM_GPIO_LEDS) {
+        return;
+    }
+
+    gpio_led_status[id] = status;
+    
+    demo_printf("GPIO LED status: ");
+    
+    for (n = 0; n < NUM_GPIO_LEDS; n++) {
```
Comment (rasendubi) [inline common/c/targets/posix/target_gpio_led.c]: (minor) Inline `n` declaration.

File: common/c/targets/posix/target_gpio_led.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#include "target.h"
+#include "target_gpio_led.h"
+
+static bool gpio_led_status[NUM_GPIO_LEDS];
+
+void target_gpio_led_init(void)
+{
+    int n;
+    
+    for (n = 0; n < NUM_GPIO_LEDS; n++) {
```
Comment (rasendubi) [inline common/c/targets/posix/target_gpio_led.c]: (minor) Inline `n` declaration.

File: common/c/targets/posix/CMakeLists.txt [hunk 4]
```diff
@@ -15,5 +15,14 @@
...
 
 # Target support library that provides abstractions for posix target.
-add_library(target_support INTERFACE)
-target_include_directories(target_support INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+add_library(target_support 
+	    STATIC
+	    target_gpio_led.c)
+
+target_include_directories(target_support PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_include_directories(target_support PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../common)
```
Comment (rasendubi) [inline common/c/targets/posix/CMakeLists.txt:23]: Shouldn't this be private?

File: common/c/targets/posix/target_gpio_led.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        gpio_led_status[n] = false;
```
Comment (rasendubi) [inline common/c/targets/posix/target_gpio_led.c:28]: `gpio_led_status` is a static variable and is initialized with zeros by default.

File: gpiocontrol/source/c/build.sh [hunk 6]
```diff
@@ -75,10 +75,10 @@ select_arch() {
         help
         ;;
     posix)
-        echo "posix platform is not supported by this demo"
-        exit 0
+        KAA_TARGET=$target
+        KAA_PRODUCE_BINARY=true
```
Comment (rasendubi) [inline gpiocontrol/source/c/build.sh]: Why we need `.bin` files for POSIX?

File: gpiocontrol/source/c/CMakeLists.txt (hunk 1 above)
Comment (vasner) [reply gpiocontrol/source/c/CMakeLists.txt]: Fixed

File: common/c/targets/posix/target_gpio_led.c (hunk 3 above)
Comment (vasner) [reply common/c/targets/posix/target_gpio_led.c]: Fixed.

File: common/c/targets/posix/target_gpio_led.c (hunk 5 above)
Comment (vasner) [reply common/c/targets/posix/target_gpio_led.c:28]: Fixed

File: common/c/targets/posix/target_gpio_led.c (hunk 2 above)
Comment (vasner) [reply common/c/targets/posix/target_gpio_led.c]: Fixed.

File: common/c/targets/posix/CMakeLists.txt (hunk 4 above)
Comment (vasner) [reply common/c/targets/posix/CMakeLists.txt:23]: I have issue with PRIVITE. In this case it cannot find headers inside specific platform folder

File: gpiocontrol/source/c/build.sh (hunk 6 above)
Comment (vasner) [reply gpiocontrol/source/c/build.sh]: We do not. Fixed

File: gpiocontrol/source/c/CMakeLists.txt [hunk 7]
```diff
@@ -47,9 +47,11 @@ else()
 endif()
 
 # For most embedded targets binary stripping is required.
-if(${KAA_PRODUCE_BINARY})
-    add_custom_command(TARGET ${APP_NAME} POST_BUILD
-            COMMAND ${CMAKE_OBJCOPY} -O binary ${APP_NAME} ${APP_NAME}.bin)
+if("${KAA_PLATFORM}" STREQUAL "posix")
```
Comment (rasendubi) [inline gpiocontrol/source/c/CMakeLists.txt]: oops. I've missed this disables producing binaries for all non-posix platforms. (Basically, all platforms we want binaries for.)

Comment (vasner) [reply gpiocontrol/source/c/CMakeLists.txt]: Fixed.

**Review by rasendubi — APPROVED** [review]

**Review by Slash32 — APPROVED** [review]

**Pull request merged**